jsapp/js/account/plan.component.tsx

```diff
@@ -92,21 +92,26 @@ export default function Plan() {
   const [shouldRevalidate, setShouldRevalidate] = useState(false);
   const [searchParams] = useSearchParams();
   const didMount = useRef(false);
-  const hasActiveSubscription = useMemo(() => {
-    if (state.subscribedProduct) {
-      return state.subscribedProduct.some((subscription: BaseSubscription) =>
-        activeSubscriptionStatuses.includes(subscription.status)
-      );
-    }
-    return false;
-  }, [state.subscribedProduct]);
 
   const isDataLoading = useMemo(
     (): boolean =>
       !(state.products && state.organization && state.subscribedProduct),
     [state.products, state.organization, state.subscribedProduct]
   );
 
+  useEffect(() => {
+    if (
+      state.subscribedProduct !== null &&
+      state.subscribedProduct.length > 0
+    ) {
+      const subscribedFilter =
+        state.subscribedProduct[0].items[0].price.human_readable_price.split(
+          '/'
+        )[1];
+      dispatch({type: subscribedFilter});
+    }
+  }, [state.subscribedProduct]);
```

Comment:
This could be a computed function, right?

Reply:
Instead of using `price.human_readable_price` to figure out whether a subscription is monthly or yearly, you should probably use [`price.recurring.interval`](https://stripe.com/docs/api/prices/object#price_object-recurring-interval_count). Otherwise, this function would need to be rewritten if we changed how we display prices (or if `dj-stripe` changes how they compute `human_readable_price`.)

Reply:
Synced with Jessica on it, after some research dj-stripe does not offer us such an easy bridge currently but we are in the works of a new approach.